assets/js/blocks/collection-filters/utils.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { ProductCollectionQuery } from '@woocommerce/blocks/product-collection/types';
+
+function taxonomyMapper( taxonomy: string ) {
+	const map: Record< string, string > = {
+		product_tag: 'tag',
+		product_cat: 'cat',
+	};
+
+	return taxonomy in map ? map[ taxonomy ] : `_unstable_tax_${ taxonomy }`;
+}
+
+export function formatQuery( query: ProductCollectionQuery ) {
+	if ( ! query ) {
+		return {};
+	}
+
+	const sharedParams: Array< keyof ProductCollectionQuery > = [
+		'exclude',
+		'offset',
+		'order',
+		'search',
+	];
+
+	const mappedParams: { key: keyof ProductCollectionQuery; map: string }[] = [
+		{ key: 'orderBy', map: 'orderby' },
+		{ key: 'pages', map: 'page' },
+		{ key: 'parents', map: 'parent' },
+		{ key: 'perPage', map: 'per_page' },
+		{ key: 'woocommerceStockStatus', map: 'stock_status' },
+		{ key: 'woocommerceOnSale', map: 'on_sale' },
+		{ key: 'woocommerceHandPickedProducts', map: 'include' },
+	];
+
+	return Object.assign(
+		{},
+		...sharedParams.map(
+			( key ) => key in query && { [ key ]: query[ key ] }
+		),
+		...mappedParams.map(
+			( param ) =>
+				param.key in query && { [ param.map ]: query[ param.key ] }
+		),
+		...Object.entries( query.taxQuery ).map( ( [ taxonomy, terms ] ) => ( {
+			[ taxonomyMapper( taxonomy ) ]: terms,
+		} ) ),
```

Comment:
What do you think of moving this logic into the `taxonomyMapper` function? So the function accepts a `taxQuery` and does all of this, making this logic a bit easier to read in its intent.